dll/win32/shell32/dialogs/filetypes.cpp

```diff
@@ -755,19 +765,15 @@ FileTypesDlg_InsertToLV(HWND hListView, LPCWSTR Assoc, INT iItem, PFILE_TYPE_GLO
 #else
         if (!Entry->ClassKey[0])
         {
-            HeapFree(pG->hHeap, 0, Entry);
-            RegCloseKey(hKey);
-            return NULL;
+            goto exclude;
         }
 #endif
     }
 
     Entry->EditFlags = GetRegDWORD(hKey, L"EditFlags", 0);
     if (Entry->EditFlags & FTA_Exclude)
     {
-        HeapFree(pG->hHeap, 0, Entry);
-        RegCloseKey(hKey);
-        return NULL;
+        goto exclude;
     }
```

Comment:
optional: consider excluding the curly-braces here. When dealing with heap-alloc/free it is always a desireable trait to keep functions compact, so that a lot of code fits on the same screen without scrolling.